### PR TITLE
feat(components/drawer): fix withAnimation, fix working with router

### PR DESCRIPTION
**Pull request by acateland**
**What is the problem this PR is trying to solve?**
`withAnimation` doesn't work properly
**What is the chosen solution to this problem?**
fixing the way `withAnimation` work, adding a way for the router to manage if a component should work with or without animation

withDrawer now is able to use a children drawer props withTransition={true|false}
withDrawer now is able to use a children of any type withTransition={true|false}

and if you use cmf and that withDrawer is injected by react router you can rely on props.route.state.withTransition to manage if or if not the drawer or children containing the drawer should be injected with an animation.

**Please check if the PR fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/Drawer/Drawer.component.js [hunk 1]
```diff
@@ -21,14 +23,15 @@ class DrawerAnimation extends React.Component {
...
+		const { children, withTransition, ...rest } = this.props;
+		const transitionDuration = withTransition ? DEFAULT_TRANSITION_DURATION : 0;
 		return (
 			<CSSTransition
 				{...rest}
 				onTransitionComplete={this.handleTransitionComplete}
 				defaultStyle={{ transform: 'translateX(100%)' }}
-				enterStyle={{ transform: transit('translateX(0%)', 350, 'ease-in-out') }}
-				leaveStyle={{ transform: transit('translateX(100%)', 350, 'ease-in-out') }}
+				enterStyle={{ transform: transit('translateX(0%)', transitionDuration, 'ease-in-out') }}
```
Comment (frassinier) [inline packages/components/src/Drawer/Drawer.component.js]: What's about `enterStyle={withTransition && { transform: transit('translateX(0%)', DEFAULT_TRANSITION_DURATION, 'ease-in-out') }}`

File: packages/components/src/Drawer/Drawer.component.js [hunk 2]
```diff
@@ -21,14 +23,15 @@ class DrawerAnimation extends React.Component {
...
+				leaveStyle={{ transform: transit('translateX(100%)', transitionDuration, 'ease-in-out') }}
```
Comment (frassinier) [inline packages/components/src/Drawer/Drawer.component.js]: Can we create a transition object as var?

File: packages/components/src/WithDrawer/WithDrawer.component.js [hunk 3]
```diff
@@ -27,14 +28,19 @@ function WithDrawer({ drawers, children }) {
...
-					>
-						{drawer}
-					</Drawer.Animation>
-				))}
+				{drawers &&
+					drawers.map((drawer, key) => (
+						<Drawer.Animation
+							withTransition={
+								get(drawer, 'props.withTransition', true) &&
+								get(drawer, 'props.route.state.withTransition')
```
Comment (frassinier) [inline packages/components/src/WithDrawer/WithDrawer.component.js]: nice :+1:

File: packages/components/src/Drawer/Drawer.component.js (hunk 1 above)
Comment (jsomsanith) [reply packages/components/src/Drawer/Drawer.component.js]: Isn't it better not to return `<CSSTransition>` if `withTransition` is false ?

Comment (acateland) [reply packages/components/src/Drawer/Drawer.component.js]: sadly not really, many forms with autorfocus rely on the onTransitionComplete hook, so some visual glitch don't happen.

**Pull request merged**